cmd/kubeadm/app/cmd/options/token.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/pflag"
+
+	bootstrapapi "k8s.io/client-go/tools/bootstrap/token/api"
+	kubeadmapiv1alpha2 "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha2"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+)
+
+func NewBootstrapTokenOptions() *BootstrapTokenOptions {
+	bto := &BootstrapTokenOptions{&kubeadmapiv1alpha2.BootstrapToken{}, ""}
+	kubeadmapiv1alpha2.SetDefaults_BootstrapToken(bto.BootstrapToken)
+	return bto
+}
+
+type BootstrapTokenOptions struct {
+	*kubeadmapiv1alpha2.BootstrapToken
+	TokenStr string
+}
+
+func (bto *BootstrapTokenOptions) AddTokenFlag(fs *pflag.FlagSet) {
```

Comment:
If we are trying to move complexity away from commands I think we should not expose add methods for each single flags. Instead we should define add methods for group of flags. e.g.
- `AddCoreFlags` (token, ttl)
- `AddAllFlags` (token, ttl, duration, usages, groups)

Reply:
done